server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1547,71 +1551,70 @@ public void regenerateCertificatesOf(Owner owner, String productId, boolean lazy
         }
     }
 
+
     /**
-     * Remove the given entitlement and clean up.
+     * Revokes the given set of entitlements.
      *
-     * @param entitlement entitlement to remove
-     * @param regenModified should we look for modified entitlements that are affected
-     * and regenerated. False if we're mass deleting all the entitlements for a consumer
-     * anyhow, true otherwise. Prevents a deadlock issue on mysql (at least).
+     * @param entsToRevoke entitlements to revoke
      */
+    @Override
     @Transactional
-    void removeEntitlement(Entitlement entitlement, boolean regenModified) {
-        Consumer consumer = entitlement.getConsumer();
-        Pool pool = entitlement.getPool();
-
-        // Similarly to when we add an entitlement, lock the pool when we remove one, too.
-        // This won't do anything for over/under consumption, but it will prevent
-        // concurrency issues if someone else is operating on the pool.
-        pool = poolCurator.lockAndLoad(pool);
-        consumer.removeEntitlement(entitlement);
+    public void revokeEntitlements(List<Entitlement> entsToRevoke) {
+        if (entsToRevoke.isEmpty()) {
+            return;
+        }
 
-        // Look for pools referencing this entitlement as their source
-        // entitlement and clean them up as well
+        List<Pool> poolsToDelete = poolCurator.listBySourceEntitlements(entsToRevoke);
+        
+        List<Pool> poolsToLock = new ArrayList<Pool>();
+        for (Entitlement ent: entsToRevoke) {
+            poolsToLock.add(ent.getPool());
+            
+            // If we are deleting a developer entitlement, be sure to delete the
+            // associated pool as well.
+            if (ent.getPool() != null && ent.getPool().isDevelopmentPool())
```

Comment:
@mstead I have been forward porting revoke entitlements and bumped revocation of development pool entitlement. Could you just quickly look if this handling of revocation of development pools is correct? 

Btw why is it necessary to delete the pool during the revocation?